src/util/serverUtils.ts

```diff
@@ -62,26 +69,212 @@ export async function getFirstConnectionForConsoleType(
   return first.value ?? null;
 }
 
+/**
+ * Get the pip server URL for the given port.
+ * @param port
```

Comment:
JSDoc should include `@returns`
Unsure if intentional but in JSDocs here and in promiseUtils you're leaving the parameters empty; they seem like mostly obvious params but then below you have the docs for `isConnected`/`isManaged`/`isRunning` which I thought was pretty obvious from the param name, so just seems a bit inconsistent. Not a big deal.